core/che-core-api-core/src/main/java/org/eclipse/che/api/core/rest/DefaultHttpJsonRequest.java

```diff
@@ -118,6 +121,16 @@ public HttpJsonRequest addQueryParam(@NotNull String name, @NotNull Object value
     return this;
   }
 
+  public HttpJsonRequest addRequestHeader(@NotNull String name, @NotNull String value) {
+    requireNonNull(name, "Required non-null header name");
+    requireNonNull(value, "Required non-null header value");
+    if (headers == null) {
+      headers = new ArrayList<>(DEFAULT_HEADERS_LIST_SIZE);
```

Comment:
Can you elaborate on why you have decided to have this 5 as a default value instead of letting default constructor do his job?

Reply:
I have decided not to break existing design of the class (see impl. of query params), I think we can safely remove this param for both lists. WDYT?

Reply:
as I understand it was mainly done in order to reduce the initial capacity and avoid recreating array in most cases. Maybe also typical number of query params used in che = 5 ? Anyway, probably the performance impact seems to be insufficient and we might just opt for using default capacity for both query params / headers?

Reply:
Yeah, it was definitely an optimization. And author might add a comment to that value to clarify that. 
I opt for leaving it with default settings.

Reply:
I mean header related one. Let's leave query params optimisation as is

Reply:
sounds good - PR has been updated